docs/content/preview/api/ysql/user-defined-subprograms-and-anon-blocks/language-plpgsql-subprograms/plpgsql-syntax-and-semantics/executable-section/basic-statements/doing-sql-from-plpgsql.md

```diff
@@ -0,0 +1,411 @@
+
+- The placeholder _%I_ means that the actual argument will be taken as the text of an object name—and then, according to the emergent status of the name as _common_ or _exotic_, this actual argument will replace the _%I_ placeholder, respectively, either _as is_ or tightly surrounded with double-quotes. The section **[Names and identifiers](../../../../../../names-and-identifiers/)** explains the _common name_ and the _exotic name_ notions and what these notions imply for the need for double-quoting.
+- The placeholder _%L_ means that the actual argument will be taken as the text of a literal value. The replaced _%L_ placeholder will always be surrounded by single-quotes. You must therefore take account of the data type that you know that you want by writing the appropriate typecast after _%L_.
+
+Try these simple tests—first using _%I_:
+
+```plpgsql
+select
+  (select format('select * from %I',  'employees')  as "common name"),
+  (select format('select * from %I',  'Pay Grade')  as "exotic name");
+```
+
+This is the result:
+
+```output
+       common name       |        exotic name        
+-------------------------+---------------------------
+ select * from employees | select * from "Pay Grade"
+```
+
+And next using _%L_:
+
+```plpgsql
+select
+  (select format('select %L::int',           42)    as "int literal"),
+  (select format('select %L::boolean',       true)  as "boolean literal"),
+  (select format('select %L',         'My Schema')  as "text literal");
+
+```
+
+This is the result:
+
+```output
+   int literal    |   boolean literal   |    text literal    
+------------------+---------------------+--------------------
+ select '42'::int | select 't'::boolean | select 'My Schema'
+```
+
+{{< tip title="Always use the 'format()' function to add object names or values into the text of a dynamic SQL statement" >}}
+An Internet search for "SQL injection" reports that it finds about _7 million_ hits. The risk of SQL injection is considered to be a huge security threat. You'll soon find articles with code examples that illustrate how the risk can occur. This [Wikipedia piece](https://en.wikipedia.org/wiki/SQL_injection) is a good place to start.
+
+Briefly, the risk occurs when you want to construct the text of a SQL statement programmatically and you want it to contain identifiers and values that become known only at run-time. Naïve programmers simply assemble the text using the ordinary _text_ concatenation operator—but you must _never_ do this. The examples that Internet search finds show you what can go wrong if you do this.
+
+(You can reduce the size of the hit-list to about _300 thousand_ by searching for _PostgreSQL "format()" "SQL injection"_. )
+
+You must _always_ must the _format()_ built-in SQL function when you need to define the text of a dynamic SQL statement at run-time. The example above shows you how to do this.
```

Comment:
You must _always_ use the _format()_ built-in SQL function

Reply:
Good catch. Fixed.